features/smoke_tests/sessions.feature

```diff
@@ -56,47 +56,48 @@ Scenario: Manual session control works
     And I wait for 8 seconds
     And I relaunch the app after a crash
     And I configure Bugsnag for "ManualSessionSmokeScenario"
-    And I wait to receive 4 requests
+    And I wait to receive a session
 
     # Session payload
-    Then the request is valid for the session reporting API version "1.0" for the "Android Bugsnag Notifier" notifier
-    And the payload field "sessions.0.id" is stored as the value "manual_session_id"
+    Then the session is valid for the session reporting API version "1.0" for the "Android Bugsnag Notifier" notifier
+    And the session payload field "sessions.0.id" is stored as the value "manual_session_id"
     And the session "user.id" equals "123"
     And the session "user.email" equals "ABC.CBA.CA"
     And the session "user.name" equals "ManualSessionSmokeScenario"
-    And I discard the oldest request
+
+    Then I wait to receive 3 errors
```

Comment:
Just to double check - presumably the order of these steps no longer matters? i.e. if the 3 errors are received first, then mazerunner will just wait for the session and then validate the error payloads?

Reply:
Yes, that's right, MazeRunner will capture requests of kinds as they are received, storing them in separate lists internally.  That's even if they are received before there is a step that says to receive them.  I guess we could reword the step to make that clearly, as it would be taken the other way, but I think people will understand it well enough an it'll hang together.

Reply:
Makes sense - just wanted to double check that was the case!